extensions/api/data-management/contractdefinition/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/ContractDefinitionApiController.java

```diff
@@ -69,21 +85,52 @@ public ContractDefinitionApiController(Monitor monitor) {
     public ContractDefinitionDto getContractDefinition(@PathParam("id") String id) {
         monitor.debug(format("get contract definition with ID %s", id));
 
-        return null;
+        return Optional.of(id)
+                .map(service::findById)
+                .map(it -> transformerRegistry.transform(it, ContractDefinitionDto.class))
+                .filter(Result::succeeded)
+                .map(Result::getContent)
+                .orElseThrow(() -> new ObjectNotFoundException(ContractDefinition.class, id));
     }
 
     @POST
     @Override
     public void createContractDefinition(ContractDefinitionDto dto) {
         monitor.debug("create new contract definition");
-    }
+        var transformResult = transformerRegistry.transform(dto, ContractDefinition.class);
+        if (transformResult.failed()) {
+            throw new IllegalArgumentException("Request is not well formatted");
```

Comment:
Consider covering this BAD_REQUEST case in `ContractDefinitionApiControllerIntegrationTest`